libdevcore/Result.h

```diff
@@ -0,0 +1,83 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include <string>
+
+namespace dev {
+
+/// Simple generic result that holds a value and an optional error message.
+/// Results can be implicitly converted to and created from the type of
+/// the value they hold.
+///
+/// Result<bool> check()
+/// {
+///		if (false)
+///			return Result<bool>(false, "Error message.")
+///		return true;
+/// }
+///
+///	A result can also be instantiated using one of the factory methods it provides:
+///
+/// using BoolResult = Result<bool>;
+/// BoolResult check()
+/// {
+///		if (false)
+///			return BoolResult::failure("Error message");
+///		return BoolResult::success(true);
+/// }
+///
+template <class Type>
+class Result
```

Comment:
This type does only contain public members. I'd suggest to use struct instead of class+public then.